app/constants.php

```diff
@@ -134,3 +134,5 @@
 defined('HTTP_PROXY_USERNAME') or define('HTTP_PROXY_USERNAME', '');
 defined('HTTP_PROXY_PASSWORD') or define('HTTP_PROXY_PASSWORD', '');
 defined('HTTP_VERIFY_SSL_CERTIFICATE') or define('HTTP_VERIFY_SSL_CERTIFICATE', true);
+
+defined('SERVER_NAME') or define('SERVER_NAME', $_SERVER['SERVER_NAME']);
```

Comment:
The name of this parameter should be something else because you use that generate an issuer for the TOTP.

You should also check for the existence of the key `SERVER_NAME`, it's not always available, that will generate a PHP notice in certain conditions.